### DISCUSSION TITLE
How to exclude files?

### DISCUSSION THREAD

**Discussion by buddhaCode** [ElMassimo/vite-plugin-full-reload#16] [Q&A]
Hi.
I try to exclude some files from triggering a reload. I am using `laravel-vite-plugin` which has a dependency on your plugin to do the reloading stuff. I want to exclude files ending with `-livewire.blade.php`. I tried the following config. But this doesn't work. Any idea why?

```
    plugins: [
        laravel({
            input: [
                'resources/js/app.js',
            ],
            refresh: ['resources/views/**', '!(resources/views/**/*-livewire.blade.php)', '!(resources/views/**-livewire.blade.php)'],
        }),
    ]
```

The page reloads, whether I change `resources/views/dashboard.blade.php` or `resources/views/counter-livewire.blade.php`. I want it to not reload on `resources/views/counter-livewire.blade.php`.

Best, Arne

**Comment by ElMassimo** [comment]
Hi Arne!

This library [uses `picomatch`](https://github.com/ElMassimo/vite-plugin-full-reload/blob/main/src/index.ts#LL54C28-L54C28) to match files, which does support negative patterns.

The problem is that when providing several paths, it will treat it as match if _any_ of them matches.

In the example you provided, `resources/views/**` will match any view, so it doesn't matter if other paths are using a [negative glob pattern](https://github.com/micromatch/picomatch#advanced-globbing).

You could try using something like:

```ts
 refresh: ['resources/views/**!(-livewire.blade.php)'],
```

**Reply by buddhaCode** [reply]
Hi Máximo,
Thanks you for the clarification. Your suggested pattern doesn't work either. So sad. I will try to find another way. Thank you.
Best, Arne
